x-pack/platform/plugins/shared/alerting/server/rules_settings/alert_deletion/rules_settings_alert_deletion_client.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import Boom from '@hapi/boom';
+import {
+  Logger,
+  SavedObjectsClientContract,
+  SavedObject,
+  SavedObjectsErrorHelpers,
+} from '@kbn/core/server';
+import {
+  RulesSettings,
+  RulesSettingsModificationMetadata,
+  RulesSettingsAlertDeletion,
+  RulesSettingsAlertDeletionProperties,
+  DEFAULT_ALERT_DELETION_SETTINGS,
+  RULES_SETTINGS_SAVED_OBJECT_TYPE,
+  RULES_SETTINGS_ALERT_DELETION_SAVED_OBJECT_ID,
+} from '../../../common';
+import { retryIfConflicts } from '../../lib/retry_if_conflicts';
+import { alertDeletionSchema } from '../schemas';
+
+export interface RulesSettingsAlertDeletionClientConstructorOptions {
+  readonly logger: Logger;
+  readonly savedObjectsClient: SavedObjectsClientContract;
+  readonly getModificationMetadata: () => Promise<RulesSettingsModificationMetadata>;
+}
+
+export class RulesSettingsAlertDeletionClient {
+  private readonly logger: Logger;
+  private readonly savedObjectsClient: SavedObjectsClientContract;
+  private readonly getModificationMetadata: () => Promise<RulesSettingsModificationMetadata>;
+
+  constructor(options: RulesSettingsAlertDeletionClientConstructorOptions) {
+    this.logger = options.logger;
+    this.savedObjectsClient = options.savedObjectsClient;
+    this.getModificationMetadata = options.getModificationMetadata;
+  }
+
+  public async get(): Promise<RulesSettingsAlertDeletion> {
+    const rulesSettings = await this.getOrCreate();
+    if (!rulesSettings.attributes.alertDeletion) {
+      this.logger.error('Failed to get alert deletion rules setting for current space.');
+      throw new Error(
+        'Failed to get alert deletion rules setting for current space. Alert deletion settings are undefined'
+      );
+    }
+    return rulesSettings.attributes.alertDeletion;
+  }
+
+  public async update(newAlertDeletionProperties: RulesSettingsAlertDeletionProperties) {
+    return await retryIfConflicts(
+      this.logger,
+      'ruleSettingsClient.alertDeletion.update()',
+      async () => await this.updateWithOCC(newAlertDeletionProperties)
+    );
+  }
+
+  private async updateWithOCC(newAlertDeletionProperties: RulesSettingsAlertDeletionProperties) {
+    try {
+      alertDeletionSchema.validate(newAlertDeletionProperties);
+    } catch (e) {
+      this.logger.error(
+        `Failed to verify new alert deletion settings properties when updating. Error: ${e}`
+      );
+      throw e;
+    }
+
+    const { attributes, version } = await this.getOrCreate();
+    if (!attributes.alertDeletion) {
+      throw new Error('Alert deletion settings are undefined');
```

Comment:
This message appears to be duplicated below.  We should use unique messages so we can tell where the code broke when we see the log message :-)

Reply:
Updated in https://github.com/elastic/kibana/pull/211046/commits/14b6fff6ed6ce36063b4f377d54711c6f0e6c69c